### PR TITLE
Hypersonic: New version 1.0.0 

### PR DESCRIPTION
**Pull request by tonimasc**
## Description

- Update Raycast to "@raycast/api": "1.44.0",
- Update Raycast to "@raycast/utils": "1.4.11"
- Added Fuse to improve the creation of todos with tags, projects, assignees
- Added chrono-node to improve the creation of todos with due dates
- Added support for multiple database relationships
- New preferences configurator to facilitate the database connection
- New multi filter option to filter todos by projects, tags and assignees
- Updated actions commands for easy access to the new features
- Replaced snooze command with a new due date command
- Deprecated move command
- Updated menu bar command to work with the new features
- Included Notion logo in Notion related commands
- Added instant refresh to menu bar command through launchCommand method
- Updated menu bar extensions background fetch time from 1 minute to 5 minutes
- Added share task command
- Logo update

## Screencast
203420468-4f1c9fe3-668b-46cc-b082-81a0ae968d7c

203420292-280741ab-a368-48cb-8752-d2cdc8a8346b

203420121-22eb93e6-da79-45af-a9fd-d9c4195ddc9a

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by prrrcl — APPROVED** [review]

**Comment by thomaspaulmann** [comment]
This is huge!!!

File: extensions/hypersonic/src/features/todo-list/components/general-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        title="Open Extension Preferences"
+        icon={Icon.Gear}
+        onAction={openCommandPreferences}
+        shortcut={{ modifiers: ['cmd'], key: ',' }}
+      />
+      <Action.CopyToClipboard
+        title="Drop us a Line"
+        icon={Icon.Heart}
+        content={'hypersonic@reboot.studio'}
+        shortcut={{ modifiers: ['cmd', 'shift'], key: 'm' }}
```
Comment (peduarte) [inline extensions/hypersonic/src/features/todo-list/components/general-actions.tsx:30]: I'm confused by this action. I was expecting it to open a webpage where I could reach out to you, but instead it copies the email address. Im not sure it's very intuitive. 

How about open your Twitter?

**Review by peduarte — COMMENTED** [review]
First of all... WOW! I love all the improvements in Hypersonic v1. Can't wait to start using it. 

Thanks for taking the time to improve Hypersonic and continue to be pushing the boundaries of how powerful Raycast Extensions can be.

I've taken some time to go through everything, and provide a mix of feedback that
- Is inline with our guidelines
- Will help with onboarding
- Will hopefully try to improve some details even more

Some are listed below, and others are commented inline. 



---

**Oboarding**
I love the idea of Onboarding! I know this already exists in the current version of Hypersonic, and happy that you are keeping it in v1. However, after using it for the first time, I got a bit confused with some things:
- Some tasks, when "done", they disappear. Eg: _"👋 Hey there! - Press ↵ to complete your first task"_ When I press ⏎ the task disappears
- This means, that as a new user, my expectation is now that when I do something right, the item is supposed to disappear
- But then, the second task says _"🎯 Press ⌘ + D to add a due date"_ and when I complete this task, the item remains in place. My initial reaction here was that I did something wrong. 
- This pattern continues and happens a couple of more times. Eg: _"🗑️ Press ⌘ + ⌫ to delete this task"_ vs _"🏷️ Press ⌘ + L to add a label of your choice"_

To solve that, I thought that a _consistent behaviour_ to communicate to users they "did the right thing", we could update the List Item `icon` to a check `Icon.CheckCircle` and make it green.

Here's what I mean:
(ಠ ›ಠ) — @peduarte — MOyVVX4T000218@2x

I appreciate this may introduce a bit more work, and maybe duplication of actions, but I think the Onboarding is key for the first impression of Hypersonic. Let me know what you think!

---

**"Create Label" in Onboarding flow**
Currently, selecting the "Create" Action closes Raycast.
Based on _Line 37 of `set-todo-label-action.tsx`_ it's supposed to open the browser. But since the onboarding data has no `url`, nothing happens. 

Im thinking that it'd be better if we don't mount the "Create" action during the Onboarding flow. What do you think?

image

---

**Optimistic Update**
When marking a task as In Progress, it takes a while until the task is sorted to its right position. Could we optimistically update it?

https://user-images.githubusercontent.com/372831/204319346-a5828dd2-a055-4526-a396-e00471390fd8.mp4

---

**Menu Bar Command**
How do you feel about truncating the title? I find it a bit odd when the Menu looks so long:
(ಠ ›ಠ) — @peduarte — 7gYsNrP6000229@2x

This is what it looks like truncated to 40 chars. 
(ಠ ›ಠ) — @peduarte — IWotRc3I000230@2x

File: extensions/hypersonic/src/features/todo-list/hooks/use-todo-list.ts [hunk 2]
```diff
@@ -113,119 +137,242 @@ export function useTodoList() {
...
       })
+
+      await showToast(Toast.Style.Success, `Scheduled for ${name}`)
     } catch (e: any) {
       showToast(Toast.Style.Failure, e?.message)
     }
   }
 
-  const handleMoveDown = async (fromIndex: number) => {
+  const handleDelete = async (todoId: string) => {
```
Comment (pernielsentikaer) [inline extensions/hypersonic/src/features/todo-list/hooks/use-todo-list.ts:200]: I would add a `confirmAlert` in this function like this instead to ensure nothing wrong is deleted:

```
  const handleDelete = async (todoId: string) => {
    if (await confirmAlert({ title: 'Are you sure you want to delete?' })) {
      try {
        await mutate(deleteTodo(todoId), {
          optimisticUpdate(data) {
            if (!data) return data
            return data.filter((t) => t.id !== todoId)
          },
          shouldRevalidateAfter: false,
        })

        showToast(Toast.Style.Success, 'Task Deleted')
        refreshMenuBar()
      } catch (e: any) {
        showToast(Toast.Style.Failure, e?.message)
      }
    }
  }
```

File: extensions/hypersonic/src/features/todo-list/components/general-actions.tsx (hunk 1 above)
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/components/general-actions.tsx:30]: Solved https://github.com/raycast/extensions/pull/3644/commits/a371988c42c5066c8f78cf0e6bd896346e1b950e

File: extensions/hypersonic/src/features/todo-list/todo-list.tsx [hunk 3]
```diff
@@ -1,229 +1,200 @@
...
-      {data && data.length === 0 ? (
-        <EmptyList notionDbUrl={notionDbUrl} />
-      ) : (
-        <TransparentEmpty />
-      )}
+                ) : null}
+                <CopyToDoAction todo={todo} />
+                <CopyTaskLinkAction todo={todo} />
+                <Action.OpenInBrowser
+                  title="Open Detail"
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/todo-list.tsx]: Solved https://github.com/raycast/extensions/pull/3644/commits/15c3c3793ba799ba74284e07a57ae86602a5d42a

File: extensions/hypersonic/src/features/todo-list/todo-list.tsx [hunk 4]
```diff
@@ -1,229 +1,200 @@
...
-                  <ReauthorizeAction />
-                  <Action
-                    title="Open Extension Preferences"
-                    icon={Icon.Gear}
-                    onAction={openCommandPreferences}
-                    shortcut={{ modifiers: ['cmd'], key: ',' }}
+                )}
+                {todo.contentUrl ? (
+                  <Action.OpenInBrowser
+                    title="Open Link"
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/todo-list.tsx]: Solved https://github.com/raycast/extensions/pull/3644/commits/1685c1cc53f9189893d0ac74aafb3485e75521bb

File: extensions/hypersonic/src/features/onboarding/onboarding.tsx [hunk 5]
```diff
@@ -18,15 +19,13 @@ export function Onboarding() {
...
   } = useOnboarding()
 
   return (
     <List
       searchText={searchText}
       onSearchTextChange={setSearchText}
-      searchBarPlaceholder="Filter or create to-do"
+      searchBarPlaceholder="Search or create task"
     >
       {searchText ? (
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/onboarding/onboarding.tsx]: Solved https://github.com/raycast/extensions/pull/3644/commits/515c6160561ab5e246dca24f978851cc222d305b

File: extensions/hypersonic/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,24 @@
...
+- Updated actions commands for easy access to the new features
+- Replaced snooze command with a new due date command
+- Deprecated move command
+- Updated menu bar command to work with the new features
+- Included Notion logo in Notion related commands
+- Added instant refresh to menu bar command through launchCommand method
+- Updated menu bar extensions background fetch time from 1 minute to 5 minutes
+- Added share task command
+- Logo update
+
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/CHANGELOG.md:31]: Solved https://github.com/raycast/extensions/pull/3644/commits/505656113c1895a09a15981707131eef8553321a

File: extensions/hypersonic/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,24 @@
 # Hypersonic Changelog
 
+## [1.0.0] - 2022-11-21
+
+- Update Raycast to "@raycast/api": "1.44.0",
+- Update Raycast to "@raycast/utils": "1.4.11"
+- Added Fuse to improve the creation of todos with tags, projects, assignees
+- Added chrono-node to improve the creation of todos with due dates
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/CHANGELOG.md]: Solved https://github.com/raycast/extensions/pull/3644/commits/505656113c1895a09a15981707131eef8553321a

File: extensions/hypersonic/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,24 @@
 # Hypersonic Changelog
 
+## [1.0.0] - 2022-11-21
+
+- Update Raycast to "@raycast/api": "1.44.0",
+- Update Raycast to "@raycast/utils": "1.4.11"
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/CHANGELOG.md]: Solved https://github.com/raycast/extensions/pull/3644/commits/505656113c1895a09a15981707131eef8553321a

File: extensions/hypersonic/src/features/onboarding/onboarding.tsx [hunk 9]
```diff
@@ -36,63 +35,42 @@ export function Onboarding() {
...
+                    title="Open Link"
                     icon={Icon.Link}
                     url={todo.contentUrl}
                     shortcut={{ modifiers: ['cmd'], key: 'e' }}
                   />
                 ) : null}
                 <CopyToDoAction todo={todo} />
+                <DeleteTodoAction todo={todo} onDelete={handleDelete} />
               </ActionPanel.Section>
               <AuthorizationAction />
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/onboarding/onboarding.tsx]: In the new onboarding concept we removed the delete action, so this don't apply any more.

File: extensions/hypersonic/src/features/todo-list/todo-list.tsx [hunk 10]
```diff
@@ -1,229 +1,200 @@
...
+            </ActionPanel>
+          }
+        />
+      ) : null}
+      {todos?.map((todo) => (
+        <List.Item
+          key={todo.id}
+          icon={{
+            source: getProgressIcon(todo.inProgress ? 0.5 : 0),
+            tintColor: todo.inProgress ? Color.Yellow : Color.SecondaryText,
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/todo-list.tsx:127]: For now we will stick with one color, the mapping idea is great but we prefer to do it in future versions.

Regarding the yellow color we tried them all, and is the color that feels better for us. Also i noticed that the color represented in the screenshots differs from what we see in our screens.

Screenshot 2022-11-29 at 20 05 45

File: extensions/hypersonic/src/features/todo-list/components/in-progress-action.tsx [hunk 11]
```diff
@@ -17,9 +17,8 @@ export function InProgressAction({
           dark: getProgressIcon(0.5, '#ffffff'),
         },
       }}
-      title={'Set in Progress'}
+      title={'Mark as In Progress'}
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/components/in-progress-action.tsx:20]: Great idea applied here https://github.com/raycast/extensions/pull/3644/commits/6ea26c3a8eda5f870ffc56eb2ede4b01011e5592

File: extensions/hypersonic/src/features/todo-list/components/set-todo-user-action.tsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+type SetLabelActionProps = {
+  todo: Todo
+  users: User[]
+  onSetUser: (todo: Todo, user: User | null) => void
+}
+
+export function SetUserAction({ todo, users, onSetUser }: SetLabelActionProps) {
+  return (
+    <ActionPanel.Submenu
+      title="Add User"
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/components/set-todo-user-action.tsx]: Much better https://github.com/raycast/extensions/pull/3644/commits/d49dd82d2cf0dc6266cc01c4fbaea8b92b248d33

File: extensions/hypersonic/src/features/todo-list/todo-list.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/todo-list.tsx]: Didn't know it was possible to get all the apps installed in the user device. This is great 🚀 I love it compared to the checkbox in preferences.

File: extensions/hypersonic/src/features/todo-list/hooks/use-todo-list.ts (hunk 2 above)
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/hooks/use-todo-list.ts:253]: Cool 😎 it feels better integrated than I thought. It happened to me sometimes to delete a task by accident.

File: extensions/hypersonic/CHANGELOG.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/CHANGELOG.md:31]: It feels much cleaner know 🤩

File: extensions/hypersonic/src/features/onboarding/onboarding.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/onboarding/onboarding.tsx]: The onboarding concept is 🔥

File: extensions/hypersonic/src/features/todo-list/todo-list.tsx [hunk 13]
```diff
@@ -1,229 +1,200 @@
...
-                <ActionPanel>
-                  <CompleteTodoAction todo={todo} onComplete={handleComplete} />
-                  <RemindAction
-                    todo={todo}
-                    onSetDate={handleSetDate}
-                    selectTask={setTodoToEdit}
-                  />
+      {!newTodo?.title && filterCount.amount > 0 ? (
+        <List.Item
+          title={''}
```
[the remark this answers is not in the document]
Comment (tonimasc) [reply extensions/hypersonic/src/features/todo-list/todo-list.tsx:95]: I understand the importance of following the guidelines to improve consistency and usability. 

But in this case, after trying both options, the subtitle one feels better integrated and adds more differentiation between the filter and the todos.

When the title is applied feels a bit overwhelming.

Of course, we can modify this if it's not allowed, but we prefer not to.

Screenshot 2022-11-29 at 20 30 14

Screenshot 2022-11-29 at 20 31 31

**Comment by tonimasc** [comment]
Hey guys thanks for the feedback, we specially loved the onboarding concept. In general everything feels more solid now 💪.  

Let me know any other suggestions.

File: extensions/hypersonic/src/features/todo-list/hooks/use-todo-list.ts (hunk 2 above)
Comment (pernielsentikaer) [reply extensions/hypersonic/src/features/todo-list/hooks/use-todo-list.ts:253]: Happens to me all the time too 😊

**Comment by peduarte** [comment]
@tonimasc Thanks for being open to feedback and addressing them so quickly. I tested it again and it's really good! Congrats 🎉🎉🎉

**Review by peduarte — APPROVED** [review]

**Comment by peduarte** [comment]
@tonimasc @igarcido95 last thing: can we update the screenshots in the metadata folder as well, so it shows up in the store properly? thanks! 
(ಠ ›ಠ) — @peduarte — 593pn8AG000265@2x

File: extensions/hypersonic/package.json [hunk 14]
```diff
@@ -5,7 +5,7 @@
   "description": "Fastest Notion to-do app ever",
   "icon": "icon-purple.png",
   "author": "reboot.studio",
-  "version": "1.0.0",
```
Comment (peduarte) [inline extensions/hypersonic/package.json]: is it v2 now?

**Pull request merged**